src/Jobs/IndexJob.php

```diff
@@ -71,12 +89,29 @@ public function getJobType()
      */
     public function process()
     {
+        // It is possible that this Job is queued with no documents to be updated. If so, just mark it as complete
+        if ($this->totalSteps === 0) {
+            $this->isComplete = true;
+
+            return;
+        }
+
+        $remainingDocuments = $this->remainingDocuments;
+        // Splice a bunch of Documents from the start of the remaining documents
+        $documentToProcess = array_splice($remainingDocuments, 0, $this->batchSize);
+
+        $indexer = Indexer::create($documentToProcess, $this->method, $this->batchSize);
```

Comment:
```suggestion
        // Indexer is being instantiated in process() rather that __construct() to prevent the following exception:
        // Uncaught Exception: Serialization of 'CurlHandle' is not allowed
        // The CurlHandle is created in a third-party dependency
        $indexer = Indexer::create($documentToProcess, $this->method, $this->batchSize);
```